src/olympia/addons/templates/addons/impala/details.html

```diff
@@ -322,28 +322,30 @@ <h4 class="author">{{ _('by') }} {{ users_list(addon.listed_authors) }}</h4>
       </div>
     </div>
   {% endif %}
-  {% if waffle.switch('webext-permissions') and addon.current_version and addon.current_version.all_files[0] %}
+  {% if waffle.switch('webext-permissions') and
+        addon.current_version and addon.current_version.all_files[0] and
+        (not addon.current_version.all_files[0].is_webextension or
+         addon.current_version.all_files[0].webext_permissions)
```

Comment:
Since this is starting to be quite a big condition... how about moving it to a model method like `addon.should_display_permissions_dialog()` or something like that? It could even be re-used for the button code above if you allow passing an optional version parameter.